fastlane/Fastfile

```diff
@@ -879,54 +878,80 @@ def simulator_version
   '14.4'
 end
 
-def occupy_test_account
-  UI.message("Looking for a free test account...")
-  find_free_test_account
-  change_test_account_availability("T")
-end
-
 def deoccupy_test_account
   return if $used_test_account_index == nil
-  UI.message("Freeing used test account")
-  change_test_account_availability("F")
+  UI.message("Freeing used test account #{$used_test_account_index}")
+  change_test_account_availability("free")
   $used_test_account_index = nil
 end
 
-# Returns the value of test accounts key, which is 
-# expected to look like a series of Fs and Ts.
+# Test accounts info is stored in JSON-formatted string, where keys represent
+# the account index, and values are either "free" or CIRCLE_BUILD_NUM, in case if account
+# was occupied by a build. Returns the hash.
 # Example:
-#   get_test_accounts_state
-#     => "TFFFF"
-def get_test_accounts_state
-  uri = URI.parse(get_required_env("UI_TESTS_ACCOUNTS_URL"))
+#   get_test_accounts_hash
+#     => {"0"=>"4079", "1"=>"free", "2"=>"free", "3"=>"free"}
+def get_test_accounts_hash
+  uri = URI.parse(get_required_env("UI_TESTS_ACCOUNTS_JSON_URL"))
   response = Net::HTTP.get_response(uri)
   accounts_state = response.body.chomp
-  UI.message("State of test accounts: #{accounts_state}")
-  accounts_state
+  JSON.parse(accounts_state)
 end
 
-# Returns the index of first "F" (Free) occurence in the test
-# accounts keyvalue, or crashes fastlane if it was not found.
-# Example:
-#   find_free_test_account
-#     => 2
+# Finds the index of first free account in accounts hash, and marks it as accupied by current build.
+# Crashed fastlane if free account was not found.
 def find_free_test_account
-  $used_test_account_index = get_test_accounts_state.index("F")
+  accounts_hash = get_test_accounts_hash
+  UI.message("Looking for a free test account...")
+  UI.message("Accounts state: #{accounts_hash}")
+
+  $used_test_account_index = accounts_hash.key('free')
+
   UI.user_error!("Could not find free UI Test account. Quitting.") if $used_test_account_index.nil? 
   UI.message("Free account index: #{$used_test_account_index}")
+  change_test_account_availability("#{get_required_env("CIRCLE_BUILD_NUM")}")
```

Comment:
Using `CIRCLE_BUILD_NUM` is pretty neat 👍

Reply:
Thank you Gio! 🎆